html/js/tree/Tree.js

```diff
@@ -122,52 +136,57 @@ GTE.TREE = (function (parentModule) {
             this.updateLeavesPositions();
         } else {
             for (var i = 0; i < numberLeaves; i++) {
-                this.leaves[i].x = (widthPerNode*i)+(widthPerNode/2) -
+                this.leaves[i].node.x = (widthPerNode*i)+(widthPerNode/2) -
                                         GTE.CONSTANTS.CIRCLE_SIZE/2 + offset;
             }
         }
     };
 
     /**
-    * Adds a child to a given node
-    * @param  {Node} parentNode Node that will get a new child
-    * @return {Node} newNode    Node that has been added
+    * Adds a child to a given iSet
+    * @param  {ISet} parentISet ISet that will get a new child
+    * @return {ISet} newISet    ISet that has been added
     */
-    Tree.prototype.addChildNodeTo = function (parentNode) {
-        var newNode = new GTE.TREE.Node(parentNode);
-        if ((newNode.y + GTE.CONSTANTS.CIRCLE_SIZE) > GTE.canvas.viewbox().height) {
+    Tree.prototype.addChildISetTo = function (parentISet) {
+        // Create a new ISet
+        // TODO #15
+        var newISet = new GTE.TREE.ISet(parentISet, 1);
+        // Check none of the new nodes of the ISet is out of the canvas
+        // TODO #14
+        if ((newISet.node.y + GTE.CONSTANTS.CIRCLE_SIZE) > GTE.canvas.viewbox().height) {
             this.zoomOut();
         }
+        // Inform that tree needs to be redrawn
         this.positionsUpdated = false;
-        return newNode;
+        return newISet;
     };
 
     /**
-     * Function that deletes the node. It changes children's parent to their
-     * grandparent.
-     * @param {Node} node Node to be deleted
+     * Function that deletes the iSet children
+     * @param {ISet} iSet iSet whose children will be deleted
      */
-    Tree.prototype.deleteChildrenOf = function (node) {
+    Tree.prototype.deleteChildrenOf = function (iSet) {
         // Delete everything below every child
-        while(node.children.length !== 0){
-            this.recursiveDeleteChildren(node.children[0]);
+        while(iSet.moves.length !== 0){
+            // Send moves[0] because moves[] will decrease in size
+            this.recursiveDeleteChildren(iSet.moves[0].child);
```

Comment:
I would do instead:

    this.recursiveDeleteChildren(iSet.moves.pop().child);